### PR TITLE
LPD-40224

### PR DESCRIPTION
**Pull request by daiazius**
Ticket: https://liferay.atlassian.net/browse/LPD-40224

Hi, this problem was happening because when the portlet namespace was removed from the date parameters, the resourceURL tag stopped returning these parameters because it needs a namespace.

I solved this by moving the auditDisplayContext to init.jsp and using it to put these parameters back in the exportURL and making _getParamWithOrWithoutNamespace() accept the portlet namespace.

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:reevaluate:1297401_10627

File: portal-impl/src/com/liferay/portlet/internal/PortletURLImpl.java [hunk 1]
```diff
@@ -1520,7 +1527,7 @@ else if (_lifecycle.equals(PortletRequest.RENDER_PHASE)) {
 						}
 					}
 				}
-				else {
+				else if (_useNamespace) {
```
Comment (ChrisKian) [inline portal-impl/src/com/liferay/portlet/internal/PortletURLImpl.java:1530]: I don't believe we should be touching the private render param namespace.  Is this change required for your solution to work?

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/resources/META-INF/resources/configuration/icon/export_audit_events.jsp [hunk 2]
```diff
@@ -7,7 +7,20 @@
 
 <%@ include file="/init.jsp" %>
 
-<liferay-portlet:resourceURL id="/audit/export_audit_events" var="exportURL" />
+<liferay-portlet:resourceURL id="/audit/export_audit_events" var="exportURL">
+	<portlet:param name="endDateAmPm" value="<%= String.valueOf(auditDisplayContext.getEndDateAmPm()) %>" />
```
Comment (ChrisKian) [inline modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/resources/META-INF/resources/configuration/icon/export_audit_events.jsp:11]: Do we need to use `String.valueOf` for these params, or are `int` values ok to use?

Comment (daiazius) [reply modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/resources/META-INF/resources/configuration/icon/export_audit_events.jsp:11]: We need String.valueOf or we get a `int cannot be converted to java.lang.String`

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
See https://github.com/liferay-appsec/liferay-portal/pull/2054
